openquake/hazardlib/geo/surface/kite_fault.py

```diff
@@ -0,0 +1,754 @@
+                          pr[k, 0], pr[k, 1], pr[k, 2])
+
+            # Check distance
+            if abs(dd-tdist) > 0.1*tdist:
+                print('dd:', dd)
+                tmps = 'Error while building the mesh'
+                tmps += '\nDistances: {:f} {:f}'
+                raise ValueError(tmps.format(dd, tdist))
+
+            # Adding new points along the edge with index k
+            for j in range(ndists):
+
+                # Add new profile
+                if len(npr)-1 < laidx[k]+1:
+                    npr = add_empty_profile(npr)
+
+                # Compute the coordinates of intermediate points along the
+                # current edge
+                tmp = (j+1)*sd - rdist[k]
+                lo, la, _ = g.fwd(pl[k, 0], pl[k, 1], az12,
+                                  tmp*hdist/tdist*1e3)
+
+                tidx = np.argmin(abs(tdsts-tmp))
+                lo = ll[tidx, 0]
+                la = ll[tidx, 1]
+
+                # Fix longitudes
+                if idl:
+                    lo = lo+360 if lo < 0 else lo
+
+                # Computing depths
+                de = pl[k, 2] + tmp*vdist/hdist
+                de = deps[tidx]
+
+                npr[laidx[k]+1][k] = [lo, la, de]
+                if (k > 0 and np.all(np.isfinite(npr[laidx[k]+1][k])) and
+                        np.all(np.isfinite(npr[laidx[k]][k]))):
+
+                    p1 = npr[laidx[k]][k]
+                    p2 = npr[laidx[k]+1][k]
+                    d = distance(p1[0], p1[1], p1[2], p2[0], p2[1], p2[2])
+
+                    # Check
+                    if abs(d-sd) > 0.1*sd:
+                        tmpf = 'd: {:f} diff: {:f} tol: {:f} sd:{:f}'
+                        tmpf += '\nresidual: {:f}'
+                        tmps = tmpf.format(d, d-sd, TOL*sd, sd, rdist[k])
+                        raise ValueError(tmps)
+                laidx[k] += 1
+
+            rdist[k] = tdist - sd*ndists + rdist[k]
+            assert rdist[k] < sd
+
+    return npr
+
+
+def get_mesh_back(pfs, rfi, sd, idl):
+    """
+    Compute resampled profiles in the backward direction from the reference
+    profile and creates the portion of the mesh 'before' the reference profile.
+
+    :param list pfs:
+        Original profiles. Each profile is a :class:`numpy.ndarray` instance
+        with 3 columns and as many rows as the number of points included
+    :param int rfi:
+        Index of the reference profile
+    :param sd:
+        Sampling distance [in km] along the strike
+    :param boolean idl:
+        A flag used to specify cases where the model crosses the IDL
+    :returns:
+
+    """
+
+    # Projection
+    g = Geod(ellps='WGS84')
+
+    # Initialize residual distance and last index lists
+    rdist = [0 for _ in range(0, len(pfs[0]))]
+    laidx = [0 for _ in range(0, len(pfs[0]))]
+
+    # Create list containing the new profiles. We start by adding the
+    # reference profile
+    npr = list([copy.deepcopy(pfs[rfi])])
+
```

Comment:
Same comment as before. Also this function is too big and seems to have some part in common with the one before.